Game/Source/GDKTestGyms/GDKTestGymsGameInstance.cpp

```diff
@@ -21,19 +22,51 @@ void UGDKTestGymsGameInstance::OnStart()
 	}
 }
 
-bool UGDKTestGymsGameInstance::Tick(float DeltaSeconds)
+float UGDKTestGymsGameInstance::AddAndCalcFps(int64 NowReal, float DeltaS)
 {
-	float Alpha = 0.8f;
-	float CurrFPS = 1.0f / DeltaSeconds; // TODO - we should remove FApp::GetIdleTime() from DeltaSeconds once the NFR regex is fixed
-	AverageFPS = (Alpha * AverageFPS) + ((1.0f - Alpha) * CurrFPS);
+	int64 DeltaTicks = FTimespan::FromSeconds(DeltaS).GetTicks();
+	TicksForFPS.Push(TPair<int64, int64>(NowReal, DeltaTicks)); 
+	TickWindowTotal += DeltaTicks;
+
+	int64 MinRange = NowReal - FTimespan::FromMinutes(2.0).GetTicks();
+	
+	// Trim samples
+	int NumToRemove = 0;
+	for(int i = 0; i < TicksForFPS.Num(); i++)
+	{
+		const FPSTimePoint& TimePoint = TicksForFPS[i];
+		if (TimePoint.Key > MinRange) // Find the first valid sample
+		{
+			NumToRemove = i;
+			break;
+		}
+	}
+	if (NumToRemove > 0)
+	{
+		for (int j = 0; j < NumToRemove; j++)
```

Comment:
Omega Nit: the loop can use "i" now XD

Reply:
Also Omega Nit: TickWindowTotal -= TicksForFPS[j].Value logic could be performed in the previous loop. This would reduce ops but....... Also you kinda have this nice separation of functionality that would be broken.

I don't know which of these I prefer tbh I just wanted to point it out

Reply:
Yeah, I think it was slightly confusing when it was in the loop since the 2 branches were together and the logic for 2 things was merged